pkg/certificate/types.go

```diff
@@ -96,7 +97,37 @@ type Manager struct {
 // MRCClient is an interface that can watch for changes to the MRC. It is typically backed by a k8s informer.
 type MRCClient interface {
 	List() ([]*v1alpha2.MeshRootCertificate, error)
+	MRCEventBroker
 
 	// GetCertIssuerForMRC returns an Issuer based on the provided MRC.
 	GetCertIssuerForMRC(mrc *v1alpha2.MeshRootCertificate) (Issuer, pem.RootCertificate, string, error)
 }
+
+type MRCEventType string
+type MRCEvent struct {
+	Type MRCEventType
+	// The last observed version of the MRC as of the time of this event
+	MRC *v1alpha2.MeshRootCertificate
+	// The previous version of the MRC. Only populated during updates
+	// +optional
+	OldMRC *v1alpha2.MeshRootCertificate
+}
+
+var (
+	// MRCEventAdded is the type of announcement emitted when we observe an addition of a Kubernetes MeshRootCertificate
+	MRCEventAdded MRCEventType = "meshrootcertificate-added"
```

Comment:
Wondering if we might only need update/add, and even then if we need to "know" which event it was.

ie: an add will be a replacement, so knows to move the client into the old stage

Reply:
I generally err on the side of giving subscribers too much information rather than not enough. Having the context of what action is being taken on the MRC can aid in decisionmaking